pkg/bloomgateway/bloomgateway.go

```diff
@@ -315,21 +327,63 @@ func (g *Gateway) FilterChunkRefs(ctx context.Context, req *logproto.FilterChunk
 		return req.Refs[i].Fingerprint < req.Refs[j].Fingerprint
```

Comment:
Is this sort needed other than for the bin-search inside `partitionRequest`?
- If not: I think `partitionRequest` should make the sorting.
- If yes: I think `partitionRequest` should check that the chunks are sorted. And sort them if not.

Reply:
We rely on sorted fingerprints throughout the read path. So we sort once in the beginning.